ReadMe.md
![](http://i.imgur.com/dGJesJY.png)

# Version 8.3

#### A updated version of my coderhire script, that had over 90 sold copies.

##### This version is not from coderhire!

##### This version only supports the latest DarkRP version. Find it at https://github.com/FPtje/DarkRP


##### Read the changelogs here https://github.com/samahl9709/SHUD/blob/master/Changelog.md
#Donations
Send the selected amount to my paypal samueldeanhansahlstrom@gmail.com


Every donation shows me that you wan't me to keep working on SHUD.

# Support
If you need help with SHUD send me a friend request on steam or send a email with the subject "SHUD - Help".

Email: samueldeanhansahlstrom@gmail.com

Steam: http://steamcommunity.com/id/samueldha

# Wiki
If you need any help or have any problems check the wiki before contacting me https://github.com/samahl9709/SHUD/wiki

# Errors
Report all errors at https://github.com/samahl9709/SHUD/issues

# Updates
I will try to update when i have time. Feel free to send any updates to me!
Use github to request a puch or send the update to samueldeanhansahlstrom@gmail.com.

# Credits
Samuel - Original creator of SHUD

David (aka Konth) - Updated 2.0 and 3.2

Accro Gaming Players - The idea

Chibby - Suggestions

# FAQ
###### Q: When do i need to download a new update?
###### A: In-game you can run the command 'shud' if the command return the message
###### 'SHUD is out of date!' then you should update (The message will show in server console).

###### Q: I found a error, what should i do?
###### A: Check Common errors under the wiki. If your error is not there then report it at https://github.com/samahl9709/SHUD/issues

###### Q: Can you make a custom version for my server?
###### A: Yes, if its a small change i will do it for free but bigger changes will cost a little.

###### Q: I have seen this script before you started to sell it on coderhire
###### A: I have too, i had a old version of SHUD on my old server Accro Gaming, and when i shut it down a lot of
###### players and other servers wanted to buy it so i started to sell it.

# How to install
1. Open SHUD-Master (Do not move SHUD-Master to /darkrp_modules)
1. Move SHUD to addons > darkrpmodification > lua > darkrp_modules. (The SHUD folder is inside the SHUD-Master folder)
2. Open sh_config.lua.
3. Edit the following configs as you want them.
4. Restart your server.
5. Look for errors and report them at https://github.com/samahl9709/SHUD/issues

Wan't to remove the darkrp hud over head? Press here https://github.com/samahl9709/SHUD/wiki/Remove-DarkRP-Hud

# Upcomming Update 8.6
- Support for more gamemodes
- In-Game menu for Workshop version
- Settings for custom box (For levels etc..)


Changelog.md
## Changelog

### 8.3.3 (2014/08/22
####cl_shud.lua (Client Side)

Edited if statments

####sh_config.lua (Shared Side)

Added SHUD.Config.Icon.Warning

Renamed SHUD.Config.Showwanted to SHUD.Config.DarkRP.Showwanted

Renamed SHUD.Config.Patrol to SHUD.Config.DarkRP.Patrol

Renamed SHUD.Config.Text.Wanted to SHUD.Config.DarkRP.Wanted

Renamed SHUD.Config.Gun to SHUD.Config.Icon.Gun

Renamed SHUD.Config.Paper to SHUD.Config.Icon.Paper

Change SHUD.Config.Addon.Gangsize from "small" to "medium"

Change SHUD.Config.Addon.Orgsize from "small" to "medium"

Removed non working configs

####sv_shud.lua (Server Side)

Edited Version check function

Removed non working function

### 8.3.2 (2014/08/22)
####cl_shud.lua (Client Side)

Added font size if statments (See sh_config.lua)

Added toggle if statment

Added SHUD deactivation when you are a listed class or rank

Fixed so the wanted box wont cover the gang/org box when its up

####sh_config.lua (Shared Side)

Added SHUD.Config.RHide

Added SHUD.Config.Patrol

Added SHUD.Config.Addon.GangSize

Added SHUD.Config.Addon.OrgSize

Added SHUD.Config.Addon.Showorg

Added SHUD.Config.Addon.Gangtext

Added SHUD.Config.Addon.Orgtext


Removed SHUD.Config.Dead

####sv_shud.lua (Server Side)

Added "PlayerSay" hook
Added Toggle chat command and function


### 8.3.1 (2014/08/21)
####cl_shud.lua (Client Side)

Added SHUD.Config.Second if statment

Changed some if statments

Moved local tcol = team.GetColor(ply:team()) higher up

Removed spaces

Removed local CTeam = ply:Team()



####sh_config.lua (Shared Side)
Added SHUD.Config.Second (Enable/Disable the job/health box)

Added SHUD.Config.Addon.Org (Organisations Addon for DarkRP)

Moved SHUD.Config.ShowEx higher up

Changed SHUD.Config.DEV from {"superadmin"} to "superadmin"


####sv_startup.lua (Server Side)
Added Version to the load message

sv_shud.lua (Server Side)

Added Message that tells you that the version checker is loading

Added Function that will give a rank the outdated message (Not Done)



SHUD/sv_shud.lua

///////////////////////////////////////////
//  Made by Samuel 'STEAM_0:1:42929463'  //
///////////////////////////////////////////

include('sh_config.lua')


SHUD = {}
SHUD.Config = {}
SHUD.Build = "833"
SHUD.Version = "8.3.3"
SHUD.CurrentBuild = "0"
SHUD.LatestBuild = "0"
SHUD.BuildOutdated = false

local function CompareVersions()
if tonumber(SHUD.CurrentBuild) < tonumber(SHUD.LatestBuild) then
  MsgN('')
  MsgN('SHUD is out of date!')
  MsgN('Local build: ' .. SHUD.CurrentBuild .. ', Latest build: ' .. SHUD.LatestBuild .. '')
  MsgN('Download the latest version from https://github.com/SDHAScripts/SHUD/archive/master.zip')
  MsgN('')

SHUD.BuildOutdated = true
else
  MsgN('')
  MsgN('SHUD is on the latest version.')
  MsgN('')
end
end

function CheckVersion()
  SHUD.CurrentBuild = SHUD.Build

	MsgN('SHUD is loading version checker. Please wait...')
	local url = 'https://raw.github.com/SDHAScripts/shud/master/build.txt'
	http.Fetch( url,
		function( content )
			SHUD.LatestBuild = tostring( content ) or "Error"
			CompareVersions()
		end,
		function(failCode)
			MsgN('SHUD couldn\'t check version.')
			MsgN(url, ' returned ', failCode)
		end
	)
end
concommand.Add("shud", CheckVersion)

hook.Add("PlayerSay", "SHUDPLAYER", function( ply, text, team )
	if ( string.sub( text, 1, 5 ) == "!shud" ) then
	local gettoggle = ply:GetNWString("SHUD_TOGGLE")
	
	if (gettoggle == "true") then
		ply:SetNWString("SHUD_TOGGLE", "false")
		ply:ChatPrint("SHUD is now deactivated")
	elseif (gettoggle == "false") then
		ply:SetNWString("SHUD_TOGGLE", "true")
		ply:ChatPrint("SHUD is now activated")
	else
		ply:SetNWString("SHUD_TOGGLE", "false")
		ply:ChatPrint("SHUD is now deactivated")
	end
	
	return(false)
	
	end
end )


SHUD/sh_config.lua
SHUD = {}
SHUD.Config = {}
SHUD.Config.DLC = {}
SHUD.Config.Icon = {}
SHUD.Config.Text = {}
SHUD.Config.Addon = {}
SHUD.Config.Color = {}
SHUD.Config.Ranks = {}

// General Configs
SHUD.Config.DarkRP              = true                  -- Using DarkRP?
SHUD.Config.Second              = true                  -- Show the second bar (HP and Job)
SHUD.Config.Showrank            = true                  -- Show the rank of admins
SHUD.Config.ShowEx              = "health"              -- false = disabled, Use health or armor.
SHUD.Config.Range               = 250                   -- Range of the player card
SHUD.Config.Round               = 1                     -- The main box: 1 = Default , 2 = Square
SHUD.Config.Font                = 1                     -- Text fonts: 1 = Default, 2 = SHUD BIG, 3 = Old School
SHUD.Config.Lang                = 1                     -- Languages: 1 = English, 2 = Swedish, 3 = Danish
SHUD.Config.RHide               = 'false'               -- Usergroup (SHUD will be deactivated for this person)

// DarkRP Configs
SHUD.Config.Showwanted  	      = true                  -- Show if a player is wanted
SHUD.Config.Patrol              = "TEAM_ADMIN"          -- Name of the admin class (The SHUD will be deactivated for this class)
SHUD.Config.Wanted              = "Custom Wanted Text"  -- Custom wanted text, set Lang to 4 to use this

// DarkRP Icons
SHUD.Config.Icon.Gun            = "gun"                 -- Silkicon for the gun
SHUD.Config.Icon.Paper          = "page"                -- Silkicon for the paper
SHUD.Config.Icon.Warning        = "stop"                -- Silkicon for the warning sign

// Configs for other Addons and Modules
SHUD.Config.Addon.Showgang      = false                 -- Show gang name (DarkRP Gang System) (Not 100% sure if its working)
SHUD.Config.Addon.GangSize      = "medium"              -- Text size (small/medium/large)
SHUD.Config.Addon.Showorg       = false                 -- Show org name (Organisations Addon for DarkRP)
SHUD.Config.Addon.OrgSize       = "medium"              -- Text size (small/medium/large)

// Configs for SHUD DLC's
SHUD.Config.DLC.Tag             = false                 -- Custom tag (SHUD Tag System)

// Colors (Do not change the last ',180'...)
SHUD.Config.Color.Nametext      = Color(10,10,10,180)
SHUD.Config.Color.Namebg        = Color(160,160,160)

SHUD.Config.Color.Jobtext       = Color(255,255,255)
SHUD.Config.Color.Jobbg         = Color(255,255,255)

SHUD.Config.Color.Ranktext      = Color(255,255,255,180)
SHUD.Config.Color.Rankbg        = Color(0,0,0,180)

SHUD.Config.Addon.Gangcolor     = Color(0,0,0,180)
SHUD.Config.Addon.Gangtext      = Color(255,255,255,255)
SHUD.Config.Addon.Orgcolor      = Color(10,10,10,180)
SHUD.Config.Addon.Orgtext       = Color(255,255,255,255)

// Ranks (First color code = Box, Second color code = Text)
SHUD.Config.Ranks["superadmin"] = {"Super Admin", "shield_add", Color(10,10,10,180), Color(255,255,255,180)}
SHUD.Config.Ranks["admin"]      = {"Admin", "shield", Color(10,10,10,180), Color(255,255,255,180)}


SHUD/cl_shud.lua
--[[
  Made by Samuel 'STEAM_0:1:42929463'
]]--

include('sh_config.lua')

--[[
 Version 8.3.3 2014/08/28
]]--

--[[
    Only edit below if you know what you are doing!
]]--

if (SHUD.Config.Font == 1) then
    surface.CreateFont ("SHUDName", {
    size = 23,
    weight = 400,
    antialias = true,
    shadow = false,
    font = "coolvetica"})
end

if (SHUD.Config.Font == 2) then
    surface.CreateFont ("SHUDName", {
    size = 23,
    weight = 400,
    antialias = true,
    shadow = false,
    font = "DejaVu Sans"})
end

if (SHUD.Config.Font == 3) then
    surface.CreateFont ("SHUDName", {
    size = 20,
    weight = 400,
    antialias = true,
    shadow = false,
    font = "akbar"})
end

if (SHUD.Config.Addon.OrgSize == "small") then
    surface.CreateFont ("SHUDORG", {
    size = 24,
    weight = 400,
    antialias = true,
    shadow = false,
    font = "coolvetica"})
end

if (SHUD.Config.Addon.OrgSize == "medium") then
    surface.CreateFont ("SHUDORG", {
    size = 26,
    weight = 400,
    antialias = true,
    shadow = false,
    font = "coolvetica"})
end

if (SHUD.Config.Addon.OrgSize == "big") then
    surface.CreateFont ("SHUDORG", {
    size = 32,
    weight = 400,
    antialias = true,
    shadow = false,
    font = "coolvetica"})
end

if SHUD.Config.Lang == 1 and SHUD.Config.DarkRP then
    wantedtext = "Wanted by The Police"
    goterror = "Please contact an administrator"
end

if SHUD.Config.Lang == 2 and SHUD.Config.DarkRP then
    wantedtext = "Efterlyst av Polisen"
end

if SHUD.Config.Lang == 3 and SHUD.Config.DarkRP then
    wantedtext = "Eftersøgt af politiet"
end

if SHUD.Config.Lang == 4 and SHUD.Config.DarkRP then
	wantedtext = SHUD.Config.Wanted
end

local function drb(r, x, y, w, h, col)
    draw.RoundedBox(r, x, y, w, h, col)
end

local function ddt(text, font, x, y, col, align)
    draw.DrawText(text, font, x, y, col, align or 0)
end

local function dicon(mat, x, y, w, h)
    surface.SetDrawColor( 255, 255, 255, 255 )
    surface.SetMaterial( Material(mat) )
    surface.DrawTexturedRect( x, y, w, h)
end

function DrawSHUD(ply)
    local pos = ply:EyePos()
    pos.z = pos.z + 15
    pos = pos:ToScreen()
    pos.y = pos.y - 60

	if SHUD.Config.DarkRP then
		ply.DarkRPVars = ply.DarkRPVars or {}
	end

    local RankData = SHUD.Config.Ranks[ply:GetUserGroup()] or {}
    local hasRank = (RankData[1] and true or false)
    local x, y = pos.x - 115 , pos.y + 1
    local w, h = 250, (hasRank and SHUD.Config.Showrank and 90 or 60)
    local tcol = team.GetColor(ply:Team())
    local name = ply:Nick() or "Joe Doe"
	local rankpos = 2

	if SHUD.Config.DLC.Tag and (ply:GetNWString("STag") == !"") then
		w, h = w, h + 28
	end

    if SHUD.Config.Round == 1 then
		round = 6
	elseif SHUD.Config.Round == 2 then
		round = 0
    end

	if SHUD.Config.Second then
		drb(round, x, y, w, h, RankData[3] or Color(10,10,10,180))
	elseif (SHUD.Config.Second == false) then
		drb(round, x, y, w, 32, RankData[3] or Color(10,10,10,180))
	end

    drb(round, x + 3, y + 3, w - 6, 25, SHUD.Config.Color.Nametext)
    ddt(name, "SHUDName", x + w / 2, y + 4, SHUD.Config.Color.Namebg, TEXT_ALIGN_CENTER)

	if SHUD.Config.DarkRP and SHUD.Config.Second then
		ddt(ply:getDarkRPVar("job") or "", "SHUDName", x + w / 2, y + 34, SHUD.Config.Color.Jobtext, TEXT_ALIGN_CENTER)
	end

    if (SHUD.Config.ShowEx == "health") or (SHUD.Config.ShowEx == "armor") and (SHUD.Config.Second == true)  then
    if (SHUD.Config.ShowEx == "health") then
		ex = ply:Health()
	elseif (SHUD.Config.ShowEx == "armor") then
		ex = ply:Armor()
	end

    local unit = (w - 6) / 100

    if (ex < 1) and SHUD.Config.Second then
		drb(round, x + 3, y + 32, w - 6, 25, Color(tcol.r / 2, tcol.g / 2, tcol.b / 2))
	elseif (SHUD.Config.Second == false) then
		-- Hmm?
	else
		drb(round, x + 3, y + 32, w - 6, 25, Color(tcol.r / 2, tcol.g / 2, tcol.b / 2))
		drb(round, x + 3, y + 32, math.Clamp(ex * unit, 12, w - 6), 25, tcol)
	end
    else
		drb(round, x + 3, y + 32, w - 6, 25, tcol)
    end

	if SHUD.Config.DarkRP and SHUD.Config.Second then
		ddt(ply:getDarkRPVar("job") or "", "SHUDName", x + w / 2, y + 34, SHUD.Config.Color.Jobtext, TEXT_ALIGN_CENTER)
    end

	if (SHUD.Config.Addon.GangSize == "small") then
		updown = 27
	elseif (SHUD.Config.Addon.GangSize == "medium") then
		updown = 28
	elseif (SHUD.Config.Addon.GangSize == "big") then
		updown = 30
	else
		updown = 26
	end

	--[[                                                                  ]]--
	if ply:GetNWString("Gang","") != "" and SHUD.Config.Addon.Showgang then
	    local gang = ply:GetNWString("Gang","") or "No Gang"
        drb(round, x, y - 28, w, 25, SHUD.Config.Addon.Gangcolor)
        ddt(gang, "SHUDORG", x + w / 2, y - updown, SHUD.Config.Addon.Gangtext, TEXT_ALIGN_CENTER)
    elseif (ply:GetNWBool("Org_Leader",true) == true) then
        drb(round, x, y - 28, w, 25, Color(10, 10, 10, 0))
	end
	--[[                                                                  ]]--

	if (SHUD.Config.Addon.OrgSize == "small") then
		updown = 27
	elseif (SHUD.Config.Addon.OrgSize == "medium") then
		updown = 28
	elseif (SHUD.Config.Addon.OrgSize == "big") then
		updown = 30
	else
		updown = 26
	end

	--[[                                                                  ]]--
	if SHUD.Config.Addon.Showorg and ply:GetNetworkedString("Name") != "" and SHUD.Config.DarkRP then
		local org = ply:GetNetworkedString("Name") or "No Org"
		drb(round, x, y - 28, w, 25, SHUD.Config.Addon.Orgcolor)
		ddt(org, "SHUDORG", x + w / 2, y - updown, SHUD.Config.Addon.Orgtext, TEXT_ALIGN_CENTER)
	end
	--[[                                                                  ]]--

    if hasRank and SHUD.Config.Showrank and SHUD.Config.Second then
        local rank = RankData[1]
        surface.SetFont("SHUDName")
        local tw = surface.GetTextSize(rank) + 30
        local txtPos = x + w / rankpos - tw / 2
        drb(round, txtPos, pos.y + 62, tw, 25, SHUD.Config.Color.Rankbg)
        ddt(rank, "SHUDName", txtPos + 24, y + 64, RankData[4] or SHUD.Config.Color.Ranktext)
        dicon("icon16/" .. RankData[2] .. ".png", txtPos + 4, y + 66, 16, 16)
	end

	if SHUD.Config.DLC.Tag and ply:GetNWString("STag") != "" then
	    local tagsize = 85
        local tagPos = x + w / 5 - tagsize / 2
	    local stag = ply:GetNWString("STag")
	    local stype = ply:GetNWString("STag_Type")
        drb(round, x + 3, y + 90, w - 6, 25, SHUD.Config.Color.Rankbg)
        ddt(stag, "SHUDName", tagPos + 24, y + 64, Color(218,165,32, 255))
    end

	if SHUD.Config.Hit and SHUD.Config.DarkRP then
		-- Working on this --
	end

    if ply:getDarkRPVar("HasGunlicense") and SHUD.Config.DarkRP then
        dicon("icon16/" .. SHUD.Config.Icon.Paper .. ".png", x + 7, y + 62, 23, 23)
        dicon("icon16/" .. SHUD.Config.Icon.Gun .. ".png", x + 9, y + 66, 16, 16)
    end

	if ply:getDarkRPVar("wanted") and SHUD.Config.DarkRP then
        dicon("icon16/" .. SHUD.Config.Icon.Warning .. ".png", x + 34, y + 62, 23, 23)
	end

	if SHUD.Config.Addon.Showorg and ply:GetNetworkedString("Name") != "" then
		wantedpos = 56
	elseif SHUD.Config.Addon.Showgang and ply:GetNWString("Gang","") != "" then
		wantedpos = 56
	else
		wantedpos = 30
	end

    if SHUD.Config.Showwanted and ply:getDarkRPVar("wanted") and SHUD.Config.DarkRP then
        local wy = y - wantedpos
        drb(round, x, wy, w, 25, Color(0, 0, 0, 200))
        ddt(wantedtext, "SHUDName", x + w / 2 - 1, wy + 3, Color(180, 20, 20), TEXT_ALIGN_CENTER)
    end
end

function SDisplay()
    local localply = LocalPlayer()
    local localpos = localply:GetShootPos()

	if localply:Alive() and not (localply:GetNWString("SHUD_TOGGLE") == "false") then
    for k, v in pairs(player.GetAll()) do
        local pos = v:GetShootPos()
	if (v:Team() == SHUD.Config.Patrol) or (v:GetUserGroup() == SHUD.Config.RHide) and (v:GetVehicle() == "nil") then
		-- Hmm?
	else
    if localpos:Distance(pos) < SHUD.Config.Range and v:Alive() and (localply:SteamID() != v:SteamID()) then
        local diff = pos - localpos
        local trace = util.QuickTrace(localpos, diff, localply)
    if trace.Hit and trace.Entity ~= v then return end
        DrawSHUD(v)
	end
	end
	end
	end
end
timer.Simple( 12, function() hook.Add("HUDPaint", "DrawSammyHud", SDisplay) end )
